src/Promitor.Core.Scraping/Configuration/Serialization/v1/Core/AzureMetadataDeserializer.cs

```diff
@@ -8,50 +9,37 @@ namespace Promitor.Core.Scraping.Configuration.Serialization.v1.Core
 {
     public class AzureMetadataDeserializer : Deserializer<AzureMetadataV1>
     {
-        private const string CloudTag = "cloud";
-        private const string TenantIdTag = "tenantId";
-        private const string SubscriptionIdTag = "subscriptionId";
-        private const string ResourceGroupNameTag = "resourceGroupName";
-
         public AzureMetadataDeserializer(ILogger<AzureMetadataDeserializer> logger) : base(logger)
         {
-        }
-
-        public override AzureMetadataV1 Deserialize(YamlMappingNode node)
-        {
-            var metadata = new AzureMetadataV1();
+            MapRequired(metadata => metadata.TenantId);
+            MapRequired(metadata => metadata.SubscriptionId);
+            MapRequired(metadata => metadata.ResourceGroupName);
+            MapOptional(metadata => metadata.Cloud, AzureEnvironment.AzureGlobalCloud, DetermineAzureCloud);
 
-            var azureCloud = node.GetEnum<AzureCloudsV1>(CloudTag);
-            var cloud = DetermineAzureCloud(azureCloud);
-
-            metadata.TenantId = node.GetString(TenantIdTag);
-            metadata.SubscriptionId = node.GetString(SubscriptionIdTag);
-            metadata.ResourceGroupName = node.GetString(ResourceGroupNameTag);
-            metadata.Cloud = cloud;
-
-            return metadata;
         }
 
-        private AzureEnvironment DetermineAzureCloud(AzureCloudsV1? azureCloud)
+        private object DetermineAzureCloud(string azureCloudString, IErrorReporter errorReporter, KeyValuePair<YamlNode, YamlNode> nodePair)
```

Comment:
I prefer this name if it's ok? Maybe we should keep the reporter as last parameter for all methods?
> 
```suggestion
        private object DetermineAzureCloud(string rawAzureCloud, KeyValuePair<YamlNode, YamlNode> nodePair, IErrorReporter errorReporter)
```

Reply:
Yeah that's fine. I'll change that.